File: clients/theplot/src/app/stories/model/default-vocabulary.ts
import { StoryVocabulary } from './story-vocabulary';

/** Mirrors the API's `DefaultVocabulary`. */
export const DEFAULT_VOCABULARY: Readonly<StoryVocabulary> = {
  timesOfDay: ['DAY', 'NIGHT', 'DAWN', 'DUSK', 'CONTINUOUS', 'LATER'],
  transitions: ['CUT TO', 'DISSOLVE TO', 'SMASH CUT TO', 'MATCH CUT TO', 'TIME CUT', 'FADE OUT'],
  characterKinds: [
    { label: 'Human', avatarShape: 'Circle' },
    { label: 'Robot', avatarShape: 'Square' },
    { label: 'Animal', avatarShape: 'Teardrop' },
    { label: 'Creature', avatarShape: 'Squircle' },
    { label: 'Voice', avatarShape: 'Circle' },
  ],
};

File: services/api/src/ProtoFast.Data/ThePlot/Entities/DefaultVocabulary.cs
namespace ProtoFast.Data.ThePlot.Entities;

/// <summary>The labels every story offers, ahead of its own <see cref="StoryVocabulary"/>.</summary>
public static class DefaultVocabulary
{
    public static readonly IReadOnlyList<string> TimesOfDay =
        ["DAY", "NIGHT", "DAWN", "DUSK", "CONTINUOUS", "LATER"];

    public static readonly IReadOnlyList<string> Transitions =
        ["CUT TO", "DISSOLVE TO", "SMASH CUT TO", "MATCH CUT TO", "TIME CUT", "FADE OUT"];

    public static readonly IReadOnlyList<CharacterKind> CharacterKinds =
    [
        new() { Label = "Human", AvatarShape = AvatarShape.Circle },
        new() { Label = "Robot", AvatarShape = AvatarShape.Square },
        new() { Label = "Animal", AvatarShape = AvatarShape.Teardrop },
        new() { Label = "Creature", AvatarShape = AvatarShape.Squircle },
        new() { Label = "Voice", AvatarShape = AvatarShape.Circle },
    ];
}
